nixos/modules/services/networking/charybdis.nix

```diff
@@ -51,46 +51,55 @@ in
         '';
       };
 
+      motd = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        description = "Charybdis MOTD.";
+      };
+
     };
 
   };
 
 
   ###### implementation
 
-  config = mkIf cfg.enable {
-
-    users.extraUsers = singleton {
-      name = cfg.user;
-      description = "Charybdis IRC daemon user";
-      uid = config.ids.uids.ircd;
-      group = cfg.group;
-    };
-
-    users.extraGroups = singleton {
-      name = cfg.group;
-      gid = config.ids.gids.ircd;
-    };
-
-    systemd.services.charybdis = {
-      description = "Charybdis IRC daemon";
-      wantedBy = [ "multi-user.target" ];
-      environment = {
-        BANDB_DBPATH = "${cfg.statedir}/ban.db";
+  config = mkIf cfg.enable (lib.mkMerge [
+    {
+      users.extraUsers = singleton {
+        name = cfg.user;
+        description = "Charybdis IRC daemon user";
+        uid = config.ids.uids.ircd;
+        group = cfg.group;
       };
-      serviceConfig = {
-        ExecStart   = "${charybdis}/bin/charybdis-ircd -foreground -logfile /dev/stdout -configfile ${configFile}";
-        Group = cfg.group;
-        User = cfg.user;
-        PermissionsStartOnly = true; # preStart needs to run with root permissions
-      };
-      preStart = ''
-        ${coreutils}/bin/mkdir -p ${cfg.statedir}
-        ${coreutils}/bin/chown ${cfg.user}:${cfg.group} ${cfg.statedir}
-      '';
 
-    };
+      users.extraGroups = singleton {
+        name = cfg.group;
+        gid = config.ids.gids.ircd;
+      };
 
-  };
+      systemd.services.charybdis = {
+        description = "Charybdis IRC daemon";
+        wantedBy = [ "multi-user.target" ];
+        environment = {
+          BANDB_DBPATH = "${cfg.statedir}/ban.db";
+        };
+        serviceConfig = {
+          ExecStart   = "${charybdis}/bin/charybdis-ircd -foreground -logfile /dev/stdout -configfile ${configFile}";
```

Comment:
I was thinking about adding service reload support on changes to the configuration file at some point which is similar to the problem you're trying to solve. For instance the prosody service is also doing this. This behavior makes a lot of sense for such services that are best used with long-running connections and service uptime.

I'm not sure though if charybdis supports config reloading by sending signals. One can at least manually reload the config via IRC commands as an oper.